nextcord/ext/commands/bot.py

```diff
@@ -905,6 +919,138 @@ def reload_extension(self, name: str, *, package: Optional[str] = None) -> None:
+            raise ValueError("The length of packages must match the length of extensions.")
+
+        if extras is not None and len(extras) != len(names):
+            raise ValueError("The length of extra parameters must match the length of extensions.")
+
+        packages_itr: Optional[Iterator] = iter(packages) if packages is not None else None
+        extras_itr: Optional[Iterator] = iter(extras) if extras is not None else None
+
+        loaded_extensions: List[str] = []
+
+        for extension in names:
+            cur_extra = next(extras_itr) if extras_itr is not None else None
+
+            if package is None:
+                package = next(packages_itr) if packages_itr is not None else None
+
+            try:
+                self.load_extension(extension, package=package, extras=cur_extra)
+            except Exception as e:
+                # we print the exception instead of raising it because we want to continue loading extensions
+                traceback.print_exception(type(e), e, e.__traceback__, file=sys.stderr)
+            else:
+                loaded_extensions.append(extension)
+
+        return loaded_extensions
+
+    def load_extensions_from_folder(
+        self, folder_name: str, filter: str = "*.py", ignore: Optional[List[str]] = None
+    ) -> List[str]:
+        """Loads all extensions found in a folder.
+
+        Once an extension found in a folder has been loaded and did not throw
+        any exceptions, it will be added to a list of extension names that
+        will be returned.
+
+        This function only works with cog setups where the entire cog is placed
+        into one file. If you have a different setup for your cogs, then you will
+        need to manually load them.
+
+        Parameters
+        ----------
+        folder_name: :class:`str`
+            The name (or path) of the folder to look through. This folder path is a
+            relative path based on the current folder.
+        filter: :class:`str`
+            The filter to use when looking for extensions. Defaults to ``*.py``.
+            To learn more about the syntax of the filter, check out the Python Docs
+            (https://docs.python.org/3/library/fnmatch.html#module-fnmatch).
+
+        Returns
+        --------
+        A :type:`List[str]` that contains the names of all of the extensions
+        that loaded successfully.
+
+        Raises
+        --------
+        ValueError
+            :param:`folder_name` is an absolute path, not a relative path.
+        ExtensionNotFound
+            An extension could not be imported.
+        ExtensionAlreadyLoaded
+            An extension is already loaded.
+        NoEntryPointError
+            An extension does not have a setup function.
+        ExtensionFailed
+            An extension or its setup function had an execution error.
+        """
+        path = Path(folder_name)
+        if path.is_absolute():
```

Comment:
Why does this have to be relative?

Besides, can't you just convert the path to relative from absolute

Reply:
It could be absolute, but I don't think Python would like having a package input of `.home.Person.Documents.MyCoolBot.cogs.Moderation` for example.

Reply:
I see.

Possibly could get funky and remove everything up to current directory instead? Although yea, possibly more effort then its worth

Reply:
Yeah so I think it's better to do relative for this situation
I do see where you're coming from tho

Reply:
Feel free to mark as resolved